src/utils/deploy/hash-fns.mjs

```diff
@@ -119,10 +119,14 @@ const hashFns = async (
     normalizedPath: path.basename(functionPath, path.extname(functionPath)),
     runtime,
     displayName,
+    generator,
   }))
   const fnConfig = functionZips
-    .filter((func) => Boolean(func.displayName))
-    .reduce((funcs, curr) => ({ ...funcs, [curr.name]: { display_name: curr.displayName } }), {})
+    .filter((func) => Boolean(func.displayName || func.generator))
```

Comment:
This is starting to get a bit gnarly. I think we should consider abstracting this into a new function that holds the list of properties we accept and does the normalisation.

Totally fine to do in a future PR, though.

Reply:
Taking this comment seriously but also I couldn't help but hear it as a surferdude 😆 

![gnarly dude](https://media.giphy.com/media/L6copNY61JjX2/giphy.gif)